Sources/NIOCore/BSDSocketAPI.swift

```diff
@@ -85,9 +87,15 @@ private let sysInet_pton: @convention(c) (CInt, UnsafePointer<CChar>?, UnsafeMut
 #endif
 
 #if os(Android)
+#if canImport(Android)
+let IFF_BROADCAST: CUnsignedInt = numericCast(Android.IFF_BROADCAST.rawValue)
+let IFF_POINTOPOINT: CUnsignedInt = numericCast(Android.IFF_POINTOPOINT.rawValue)
+let IFF_MULTICAST: CUnsignedInt = numericCast(Android.IFF_MULTICAST.rawValue)
+#else
 let IFF_BROADCAST: CUnsignedInt = numericCast(SwiftGlibc.IFF_BROADCAST.rawValue)
```

Comment:
Swift 5 has `canImport(Glibc)` and `SwiftGlibc` on Android, while Swift 6 no longer uses those but has `canImport(Bionic)` and `canImport(Android)` instead. Both have `os(Android)`.

To keep this code compiling for both Swift 5 and 6, I duplicated these C declarations. Whenever NIO stops supporting Swift 5, I will submit a pull to rip these out.

Reply:
Can we just guard this code in a compiler directive like `#if compiler(<=5.10)`. Since we always search for these when dropping swift versions

Reply:
OK, replaced this `canImport(Android)` and three others in this pull with `compiler(>=6.0)` instead. Whenever Swift 5 support is removed, should be easy to keep those declarations and remove these for Swift 5.